crates/gitbutler-branch-actions/src/branch.rs

```diff
@@ -425,58 +425,65 @@ pub fn get_branch_listing_details(
         .filter_map(Result::ok)
         .collect();
     let repo = ctx.repository();
-    let branches = list_branches(ctx, None, Some(branch_names.clone()))?;
-    let default_target = ctx
-        .project()
-        .virtual_branches()
-        .get_default_target()
-        .context("failed to get default target")?;
-    let mut enriched_branches = Vec::new();
-
-    let default_local_branch =
-        repo.find_branch(default_target.branch.branch(), git2::BranchType::Local)?;
-    let default_branch = default_local_branch.upstream()?;
-    let head_commit = default_branch.get().peel_to_commit()?;
+    let branches = list_branches(ctx, None, Some(branch_names))?;
+
+    let (default_target_upstream_commit_id, default_target_commit_id) = {
+        let default_target = ctx
+            .project()
+            .virtual_branches()
+            .get_default_target()
+            .context("failed to get default target")?;
+        let default_local_branch =
+            repo.find_branch(default_target.branch.branch(), git2::BranchType::Local)?;
+        let default_branch = default_local_branch.upstream()?;
+        (
+            default_branch.get().peel_to_commit()?.id(),
+            default_target.sha,
+        )
+    };
 
+    let mut enriched_branches = Vec::new();
     for branch in branches {
-        let merge_base_comparison = if let Some(virtual_branch) = branch.virtual_branch {
+        let other_branch_commit_id = if let Some(virtual_branch) = branch.virtual_branch {
```

Comment:
Could someone help me understand this logic?

I think what I fail to grasp is why applied virtual branches merge with and compare to what *probably* comes down to `refs/heads/main`, whereas unapplied branches compare to *probably* `refs/remotes/origin/main`.

Reply:
I think there is a difference in intent that is poorly expressed. 
When we are previewing branches made by others, a user would generally expect the diff to be the same diff as what a github pull request would show, which is the state of the origin branch.
On the other hand when working locally, you may not have updated the workspace base yet (because you are not ready for it). In that context it may make sense to compare to that local main. 
Let me know if this explanation doesnt quite make sense still

Reply:
Thank you, this does make sense but only if I think in terms of local branch and local tracking branch.

And I think once there are some tests it will quickly become clear what is what, and maybe then I can add the documentation that I would need to remember.